src/handlers/commands_handler.ts

```diff
@@ -59,15 +59,14 @@ export const reconnectRepoHandler = async () => {
 export const trackRepoHandler = () => {
 	let repoPath = pathUtils.getRootPath();
 	if (!repoPath) return;
-	const settings = generateSettings();
-	const config = readYML(settings.CONFIG_PATH);
-	const result = checkSubDir(repoPath);
-	if (result.isSubDir) {
-		repoPath = result.parentRepo;
+	const repoState = new RepoState(repoPath);
+	const isSubDir = RepoState.isSubDir();
```

Comment:
How does this function `isSubDir()` determine if a repo is a sub directory? as usually most of our repos are sub-directories (e.g. saad/home/documents/codesync/codesyncserver)

Reply:
You can see the implementation [here](https://github.com/codesyncapp/codesync-vs/pull/187/files#diff-3f67b26a02374084675c7abac87c6c92c7f72c1668241c6bf1d0042eb5de6a2eR41)

We need to check if Currently Opened Repo is a sub-dir of some already Connected Repo so that we can show msgs accordingly and don't let the user connect it as standalone repo. 
This PR has mostly refactoring, it is there in VSCode already.

Reply:
@saadberry Let me know if you have any more questions

Reply:
Okay, thanks
 
> We need to check if Currently Opened Repo is a sub-dir of some already Connected Repo so that we can show msgs accordingly and don't let the user connect it as standalone repo. This PR has mostly refactoring, it is there in VSCode already.

Reply:
> @saadberry Let me know if you have any more questions

Yeah just a minor question about `this.config` (and what it holds) value in setSubDirState

Reply:
we have config file as `~/.codesync/config.yml`
Regarding this.config, it's value is set in the constructore, holds YAML/JSON data of config file.